server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -151,7 +151,7 @@
     /**
      * Listeners for snapshot deletion keyed by delete uuid as returned from {@link SnapshotDeletionsInProgress.Entry#uuid()}
      */
-    private final Map<String, List<ActionListener<Void>>> snapshotDeletionListeners = new HashMap<>();
+    private final Map<String, List<ActionListener<Void>>> snapshotDeletionListeners = new ConcurrentHashMap<>();
```

Comment:
Why do we need a concurrent map here now?

Reply:
Hum that looks wrong indeed 🤔 I think I forgot to revert this change during investigation.